File: 024RadioGroup/app/src/main/java/com/example/a024radiogroup/Mainactivity.java
package com.example.a024radiogroup;

import android.os.Bundle;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class Mainactivity extends AppCompatActivity implements RadioGroup.OnCheckedChangeListener {

    RadioGroup radioGroup;
    CheckBox checkBox;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        radioGroup =findViewById(R.id.pica);
        radioGroup.setOnCheckedChangeListener(((group, checkedId) -> onCheckedChanged(group,checkedId)));
        checkBox=findViewById(R.id.checkBox);
        checkBox.setOnCheckedChangeListener(((buttonView, isChecked) -> onCheckedChanged(buttonView,isChecked)));
    }

    private void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
        TextView t = findViewById(R.id.respuesta);
    if(isChecked)
        t.setText("\nTe gusta el furbo");
    else
        t.setText("\nNo te gusta el furbo");


    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        //Toast.makeText(this,"has picado"+checkedId,Toast.LENGTH_LONG).show();
        TextView text= findViewById(R.id.respuesta);
        if (R.id.radioButton==checkedId){
            text.setText("Viva cristo");
        }else if (R.id.radioButton==checkedId){
            text.setText("Viva cristo");

        } else if (R.id.radioButton==checkedId) {
            text.setText("Viva cristo");

        }else {
            text.setText("Viva cristo");

        }

    }
}

File: 024RadioGroup/app/src/main/java/com/example/a024radiogroup/MainActivity.kt
package com.example.a024radiogroup

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import android.widget.RadioGroup
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity(), RadioGroup.OnCheckedChangeListener {
    lateinit var radioGroup: RadioGroup

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        radioGroup = findViewById(R.id.pica)
        this.radioGroup.setOnCheckedChangeListener(this)

    }

    override fun onCheckedChanged(radioGroup: RadioGroup, i: Int) {
        Toast.makeText(this, "Has picado $i", Toast.LENGTH_LONG).show()
        val t = findViewById<TextView>(R.id.respuesta)
        when (i) {
            1 -> {
                findViewById<View>(R.id.radioButton)
                t.text = "El albacete mola"
            }

            2 -> {
                findViewById<View>(R.id.radioButton2)
                t.text = "El albacete mola"
            }

            3 -> {
                findViewById<View>(R.id.radioButton)
                t.text = "El albacete mola"
            }

            4 -> {
                findViewById<View>(R.id.radioButton)
                t.text = "El albacete mola"
            }
        }
    }
}